### ISSUE TITLE
Add a filter to the user / plans page tables

### ISSUE THREAD

**Issue by sjDCC** [DMPRoadmap/roadmap#2450]
On the two admin pages for "users" and "plans" add a filter that allows admins to search statistics

The primary reason we're adding this is to enable a click through from the usage dashboard so admins can interrogate specific peaks or troughs in the data. We expect most people to simple land on the results page from the filter, rather than using the filter directly themselves.
![peaks](https://user-images.githubusercontent.com/6169940/79246846-a0a82600-7e71-11ea-94d5-5abf86dfed88.PNG)

On the two tables, we feel the filter would be best to be fairly discrete e.g. a cogwheel in the table header that opens a modal which allows you to set search or select the month from a dropdown list. We may face a couple of issues in adding this as the plans table doesn't have a "month created" column, rather "last updated". 
![filter in table](https://user-images.githubusercontent.com/6169940/79247168-0f857f00-7e72-11ea-89d0-78291a30ca53.PNG)

Another option is to add the filter as another search bar entry but we think this could get confusing and clutter up the UI so expect something in the table header would be better as the filter will only work on certain fields. @mariapraetzellis will share some example options from xls and salesforce
![cogwheel](https://user-images.githubusercontent.com/6169940/79247191-17452380-7e72-11ea-9456-c0c94d520b74.PNG)

More spec ideas to come

**Comment by mariapraetzellis** [comment]
Here's a screenshot of what I was thinking about in terms of making the columns filterable w/ the cog icon and sortable w/ the arrow. I'm not sure this would work in terms of pulling out users from a specific date range though. 
![screenshot-partner archive-it org-2020 04 14-13_54_17 (1)](https://user-images.githubusercontent.com/27511575/79273576-130c0c80-7e58-11ea-8c66-b66771d48949.png)

**Comment by briri** [comment]
spoke with @mariapraetzellis and agreed that we can update the existing search box to allow users to provide a 'month and year' or a 'year' (e.g. 'Oct 2016' or '2016').

**Comment by briri** [comment]
Here is the text that appears on the Plans and Users pages:

New message about filtering (will change this in the new ticket and include a list of fields that are searchable): "_Note: You can filter this table by 'Created' dates. Enter the month abbreviation and a 4 digit year into the search box above.<br>For example: 'Oct 2019' or 'Jun 2013'._"

New message trying to explain any discrepancies between the usage dashboard's number of plans and what the user sees in the table: "_The data on the usage dashboard is historical in nature. This means that the number of records below may not match the count shown on the usage dashboard. For example if one of your users created a plan in October and then removed that plan in November, it would have been included on the usage dashboard's total for October but would not appear in the list below._" 

New message trying to explain the discrepancies between the usage dashboard's number of users and what the user sees in the table: "_The data on the usage dashboard is historical in nature. This means that the number of records below may not match the count shown on the usage dashboard. For example if one of your users joined in October and then moved to a different organization or deactivated their account, they would have been included on the usage dashboard's total for October but would not appear in the list below._"

Let me know if the above messages should be reworded

**Comment by sjDCC** [comment]
This is really nice. Thanks @briri!

A couple of comments:

1. When you first come through to these search results they can be a bit hidden on the plans page if the org has lots of plans in review. Perhaps we need to move the plan review notification dashboard to a new page, especially as people want more functionality on this. See #2365 

2. Currently the https://dmponline-test.dcc.ac.uk/org_admin/plans is throwing a 404. Not sure if that is linked to this ticket

3. Good to keep the main note text on how to use this search visible at all times. It seems that's the case from what I can see on users page. Note about different tallies reads fine.
![note](https://user-images.githubusercontent.com/6169940/80206916-23c13d00-8625-11ea-9110-770157f570c3.PNG)

4. The clickthrough is a bit unintuitive for super-admins. We see a single orgs usage dashboard but our view of users and plans is for all orgs so the numbers don't tally at all. Maybe need to flag this for self-installs and clients like Tuuli?

**Comment by briri** [comment]
Added #2483 for the click-through from the Plans by Template chart and to address 6) above. Right now the click through uses the existing 'search' functionality built into our tables. Passing 2 search terms adds quite a bit more complexity. 

Made a note on #2365 about moving the feedback table to its own page. I think that would really simplify things 👍 

4) May be a bug due to a plan with no owner (at least I encountered this in my tests against the dmptool db). @xsrust or @raycarrick-ed can confirm if its the same issue with your data.
